src/spikeinterface/postprocessing/amplitude_scalings.py

```diff
@@ -525,133 +576,128 @@ def fit_collision(
-#     )
-#     tr_overlap = recording.get_traces(start_frame=sf, end_frame=ef, channel_ids=channel_ids, return_scaled=True)
-#     ts = np.arange(sf, ef) / recording.sampling_frequency * 1000
-#     max_tr = np.max(np.abs(tr_overlap))
-
-#     for ch, tr in enumerate(tr_overlap.T):
-#         _ = ax.plot(ts, tr + 1.2 * ch * max_tr, color="k")
-#         ax.text(ts[0], 1.2 * ch * max_tr - 0.3 * max_tr, f"Ch:{channel_ids[ch]}")
-
-#     used_labels = []
-#     for i, spike in enumerate(overlapping_spikes):
-#         label = f"U{spike['unit_index']}"
-#         if label in used_labels:
-#             label = None
-#         else:
-#             used_labels.append(label)
-#         ax.axvline(
-#             spike["sample_index"] / recording.sampling_frequency * 1000, color=f"C{spike['unit_index']}", label=label
-#         )
-
-#     if scalings is not None:
-#         fitted_traces = np.zeros_like(tr_overlap)
-
-#         all_templates = we.get_all_templates()
-#         for i, spike in enumerate(overlapping_spikes):
-#             template = all_templates[spike["unit_index"]]
-#             overlap_index = np.where(spikes == spike)[0][0]
-#             template_scaled = scalings[overlap_index] * template
-#             template_scaled_sparse = template_scaled[:, sparse_indices]
-#             sample_start = spike["sample_index"] - we.nbefore
-#             sample_end = sample_start + template_scaled_sparse.shape[0]
-
-#             fitted_traces[sample_start - sf : sample_end - sf] += template_scaled_sparse
-
-#             for ch, temp in enumerate(template_scaled_sparse.T):
-#                 ts_template = np.arange(sample_start, sample_end) / recording.sampling_frequency * 1000
-#                 _ = ax.plot(ts_template, temp + 1.2 * ch * max_tr, color=f"C{spike['unit_index']}", ls="--")
-
-#         for ch, tr in enumerate(fitted_traces.T):
-#             _ = ax.plot(ts, tr + 1.2 * ch * max_tr, color="gray", alpha=0.7)
-
-#         fitted_line = ax.get_lines()[-1]
-#         fitted_line.set_label("Fitted")
-
-#     ax.legend()
-#     ax.set_title(f"Spike {spike_index} - sample {center_spike['sample_index']}")
-#     return ax
+### Debugging ###
+def _plot_collisions(we, sparsity=None, num_collisions=None):
+    """
+    Plot the fitting of collision spikes for debugging.
+    ----------
+
+    Parameters
+    we : SortingAnalyzer
+        The SortingAnalyzer object.
+    sparsity : ChannelSparsity, default=None
+        The ChannelSparsity. If None, only main channels are plotted.
+    num_collisions : int, default=None
+        Number of collisions to plot. If None, all collisions are plotted.
+    """
+    assert we.is_extension("amplitude_scalings"), "Could not find amplitude scalings extension!"
+    sac = we.load_extension("amplitude_scalings")
+    handle_collisions = sac._params["handle_collisions"]
+    assert handle_collisions, "Amplitude scalings was run without handling collisions!"
+    scalings = sac.get_data()
+
```

Comment:
yeah this is old so probably need to convert to sorting_analyzer. (is_extension for example was already deprecated even in 100.x)

Reply:
That sounds good, maybe this PR is a good opportunity to add tests for these plotting functions (to check that the conversion to sorting_analyzer is correct) and also extend the tests for `amplitude_scaling.py`.  I am not ATM familiar with the `AnalyzerExtensioncommonTestSuite` (looks very nice tho BTW) but as far as I can tell this checks the pipeline node runs without error, with and without `handle_collisions` [here](https://github.com/JoeZiminski/spikeinterface/blob/d37c2bc646180115f9d7108a7dfedde78ab847a5/src/spikeinterface/postprocessing/tests/test_amplitude_scalings.py#L19). And without handling collisions, it checks that the median scalings are some reasonable value. I'd be happy to extend this to:

1) unit test fit and find collisions, also `_units_are_spatially_overlapping` with some generated test input against expected output
2) perform a quick smoke test on `_plot_one_collision`, `_plot_collisions` to check they run without crashing. They are only internal for debugging anyways.
3) Testing `ComputeAmplitudeScalings` outputs are as expected. I am not certain on how best to do this. Maybe something like running it with toy data, then inspecting the scalings (under different input arguments) for some random spikes to check indeed the scaling is (exactly? or close to) the linear regression of waveform onto template. I think it will be hard to test this in the case of collisions, but at least the unit tests will go reasonably far for that.